consensus/src/consensus.rs

```diff
@@ -28,18 +30,22 @@ pub struct ConsensusClient<R: ConsensusRpc, DB: Database> {
     phantom: PhantomData<R>,
 }
 
-#[derive(Debug)]
-pub struct Inner<R: ConsensusRpc> {
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ConsensusStateManager<R: ConsensusRpc> {
+    #[serde(skip)]
     rpc: R,
     store: LightClientStore,
-    last_checkpoint: Option<Vec<u8>>,
-    block_send: Sender<Block>,
-    finalized_block_send: watch::Sender<Option<Block>>,
-    checkpoint_send: watch::Sender<Option<Vec<u8>>>,
-    pub config: Arc<Config>,
+    pub last_checkpoint: Option<Vec<u8>>,
+    #[serde(skip)]
+    block_send: Option<Sender<Block>>,
+    #[serde(skip)]
+    finalized_block_send: Option<watch::Sender<Option<Block>>>,
+    #[serde(skip)]
+    checkpoint_send: Option<watch::Sender<Option<Vec<u8>>>>,
+    pub config: Config,
```

Comment:
Why did you replace `Arc<Config>` with Config?

Reply:
Arc<Config> does not support any ser/de - which we need ..